src/boolean-crosses/index.js

```diff
@@ -86,21 +86,19 @@ function doMultiPointAndLineStringCross(multiPoint, lineString) {
 }
 
 function doLineStringsCross(lineString1, lineString2) {
-    var doLinesIntersect = lineIntersect(lineString1, lineString2);
-    if (doLinesIntersect.features.length > 0) {
-        for (var i = 0; i < lineString1.coordinates.length - 1; i++) {
-            for (var i2 = 0; i2 < lineString2.coordinates.length - 1; i2++) {
-                var incEndVertices = true;
-                if (i2 === 0 || i2 === lineString2.coordinates.length - 2) {
-                    incEndVertices = false;
-                }
-                if (isPointOnLineSegment(lineString1.coordinates[i], lineString1.coordinates[i + 1], lineString2.coordinates[i2], incEndVertices)) {
-                    return true;
-                }
-            }
-        }
-    }
-    return false;
+    const intersections = lineIntersect(lineString1, lineString2);
+
+    const endpoints = [
+        lineString1.coordinates[0], lineString1.coordinates[lineString1.coordinates.length - 1],
+        lineString2.coordinates[0], lineString2.coordinates[lineString2.coordinates.length - 1]
+    ]
+
+    // A line that touches another line should not cross
+    var intersectionsWithoutEndpoints = intersections.features.filter(function (feature) {
```

Comment:
`filter` functions are generally quite slow @chriserickson (as are many of those es5 array functions like `forEach` and `map`) so generally we try to avoid them if possible.

Reply:
@rowanwins: In this case the filter is quite overly aggressive too -- we don't need the filtered results.
How do you feel about `array::some()`? According to at least one source (http://www.andygup.net/fastest-way-to-find-an-item-in-a-javascript-array/) it should be plenty fast.